boost.BUILD

```diff
@@ -0,0 +1,149 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "headers",
+    hdrs = glob(
+        [
+            "boost/**",
+        ],
+    ),
+    defines = ["BOOST_ALL_NO_LIB"],
+    include_prefix = ".",
+)
+
+cc_library(
+    name = "asio",
+    srcs = glob(
+        [
+            "libs/asio/src/**",
+        ],
+    ),
+    linkopts = select({
+        "@platforms//os:linux": ["-lpthread"],
+        "@platforms//os:windows": [],
+    }),
+    deps = [
+        ":headers",
+    ],
```

Comment:
nit: visibility is missing for most of these `cc_libraries`

Reply:
Guess it's fine for now, they are not meant for external consumption but only to satisfy vsomeips dependency